hexrd/ui/import_data_panel.py

```diff
@@ -434,21 +435,22 @@ def completed(self):
             del(detectors[det])
 
         for det in self.completed_detectors:
-            transform = detectors[det].setdefault('transform', {})
-            *zx, z = transform['tilt']
-            transform['tilt'] = (
+            *zx, z = detectors[det]['transform']['tilt']
+            detectors[det]['transform']['tilt'] = (
                 [*zx, (z + float(self.edited_images[det]['tilt']))])
-            panel_buffer = detectors[det].setdefault('panel_buffer', [])
-            panel_buffer = self.edited_images[det]['panel_buffer']
+            detectors[det]['panel_buffer'] = (
+                self.edited_images[det]['panel_buffer'])
             files.append([self.edited_images[det]['img']])
 
-        temp = tempfile.NamedTemporaryFile(delete=False, suffix='.yml')
-        data = yaml.dump(
-            self.detector_defaults['default_config'], sort_keys=False)
-        temp.write(data.encode('utf-8'))
-        temp.close()
+        temp = tempfile.NamedTemporaryFile(delete=False, suffix='.hexrd')
+        try:
+            instr = HEDMInstrument(
+                instrument_config=self.detector_defaults['default_config'])
+            instr.write_config(temp.name, style='hdf5')
+            HexrdConfig().load_instrument_config(temp.name)
+        finally:
+            Path(temp.name).unlink()
```

Comment:
Do we need to close it before we unlink it? I'm not sure...

Reply:
Looks like we need to close the file or we will get an error on Windows when we try to unlink it.

Reply:
Yes, fixed.